Added PowerShell script converted from original Python2 script

mcbobke opened the pull request:
For our work with JAMF, I was tasked with converting the JSS-LDAP-Sync Python2 script to PowerShell for ease of use in our environment. This is a direct conversion between scripting languages with some added optional logging functionality and error handling. It will require that the ActiveDirectory PowerShell module is installed, but it is likely that any Windows administrators will already have this installed.

I wanted to submit a Pull Request for this to see if it is of any use to the JAMF team to provide this for other enterprises. Please let me know if there is any additional information or changes that you would like me to provide.

Thanks for the consideration!

Commit:
Uploading PowerShell script conversion

brysontyrrell commented:
I like this. Could you update the Readme's "How to use this script" section? Put the existing instructions under a "Python Script" header and then instructions for yours under a "Powershell Script" header.

mcbobke commented:
Yes, I'll get that updated.

Commit:
Readme updates

- Updates under 'What does it do?' made to accommodate PowerShell script
- Added section 'PowerShell Script' to address specific instructions for that script

mcbobke commented:
Readme has been updated.

Commit:
Use $null to check for empty depts/buildings lists

mcbobke commented:
Pushed another quick change regarding checking for empty departments/buildings lists - an unexpected exception was thrown when iterating through an empty list.

mcbobke commented:
Thanks, @jamfit !
